File: C++/Mit Polymorphismus/Wellensittich.cpp
#include "Wellensittich.h"

#include <iostream>

void Wellensittich::fressen()
{
	cout << "Korn fressen" << endl;
}

void Wellensittich::lautGeben()
{
	cout << "zwitschern" << endl;
}

void Wellensittich::freuen()
{
	cout << "wippen" << endl;
}

Wellensittich::Wellensittich(string name)
{
	this->name = name;
}

File: Java/Mit Polymorphismus/Wellensittich.java
public class Wellensittich extends Haustier{
    public Wellensittich(){
        name = "";
    }
    public Wellensittich(String n){
        name = n;
    }

    public void fressen(){
        System.out.println("Korn fressen");
    }

    public void freuen(){
        System.out.println("wippen");
    }

    public void lautGeben(){
        System.out.println("zwitschern");
    }
}
